sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/csv/CSVFileFormat.scala

```diff
@@ -109,6 +109,20 @@ class CSVFileFormat extends TextBasedFileFormat with DataSourceRegister {
       }
     }
 
+    if (requiredSchema.length == 1 &&
+      requiredSchema.head.name == parsedOptions.columnNameOfCorruptRecord) {
+      throw new AnalysisException(
+        "Since Spark 2.3, the queries from raw JSON/CSV files are disallowed when the\n" +
+          "referenced columns only include the internal corrupt record column\n" +
+          s"(named ${parsedOptions.columnNameOfCorruptRecord} by default). For example:\n" +
```

Comment:
Here `named ${parsedOptions.columnNameOfCorruptRecord}` looks weird. As it is interpolated to configured column name in runtime. We should replace it as:

1. "named `_corrupt_record` by default", as same as in migration guide.
2. "named by the config `spark.sql.columnNameOfCorruptRecord`".

Reply:
`JsonFileFormat` has the same issue.

Reply:
As the following example uses `_corrupt_record`, let's replace it as option 1.

Reply:
Thanks @viirya. Should we also need to replace the weird part in `JsonFileFormat` with `_corrupt_record`?